habitat-lab/habitat/sims/habitat_simulator/object_state_machine.py

```diff
@@ -350,4 +361,7 @@ def get_snapshot_dict(
                     if obj_state is not None
                     else state.default_value()
                 )
-        return dict(snapshot)
+
+        self._snapshot_cache = dict(snapshot)
+        self._dirty = False
+        return dict(self._snapshot_cache)
```

Comment:
```suggestion
        return self._snapshot_cache
```
Already converted to dict at line 365, right?